nix/pkgs/dramsim3.nix

```diff
@@ -0,0 +1,13 @@
+{ stdenv, fetchFromGitHub, cmake, ninja }:
+stdenv.mkDerivation rec {
+  pname = "dramsim3";
+  version = "5a22b79fd985ee826ae8a8ddc362b01c5acdec87";
+  nativeBuildInputs = [ cmake ninja ];
+  src = fetchFromGitHub {
+    owner = "CircuitCoder";
```

Comment:
Why not using upstream?

Reply:
Various CMake and memory (UB) related problems. Basically, we cannot put MemorySystem directly in pairs and vectors if we use the upstream version, and nix installation is much more complex.

Reply:
how about maintain a patch?

Reply:
Then where do we place the patch? That still have to be tracked somewhere.

I can also just transfer it elsewhere, if there is somewhere to host it.

Reply:
> I can also just transfer it elsewhere, if there is somewhere to host it.

By 'it' I meant the repo

Reply:
we can draft PR to dramsim3, although it doesn't accept PRs...

Reply:
Resolved in 3d9f341